src/Processors/Formats/RowInputFormatWithNamesAndTypes.cpp

```diff
@@ -38,10 +40,9 @@ void RowInputFormatWithNamesAndTypes::readPrefix()
     if (getCurrentUnitNumber() != 0)
         return;
 
-    if (with_names || with_types || data_types.at(0)->textCanContainOnlyValidUTF8())
+    /// Search and remove BOM only in textual formats (CSV, TSV etc), not in binary ones (RowBinary*).
```

Comment:
It should not be matched if the text format can start with a value that can contain not only valid UTF-8 characters.

Example: String, FixedString.

```
(with_names || with_types || data_types.at(0)->textCanContainOnlyValidUTF8()
```